fe/fe-common/src/main/java/org/apache/doris/catalog/AnyStructType.java

```diff
@@ -0,0 +1,35 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.catalog;
+
+/**
+ *  Describes a AnyStructType type, used for SQL function return type,
+ *   NOT used for table column type.
+ */
+public class AnyStructType extends AnyType {
```

Comment:
is this describe whole struct type and nested element type ? why not use AnyType?

Reply:
We need specific types to describe the return field type and return struct type.
So it is the ANY_ELEMENT_TYPE and ANY_STRUCT_TYPE.
If there is only ANY_TYPE, We cannot distinguish between return field type and return struct type.